public/main.js

```diff
@@ -42,33 +65,76 @@ function reloadTodoList() {
     getTodoList(function(todos) {
         todoListPlaceholder.style.display = "none";
         var count = 0;
+        var completeList = [];
         todos.forEach(function(todo) {
+            if (!selector(todo, mode)) {
+                return;
+            }
             if (!todo.complete) {
                 count++;
             }
+            else {
+                completeList.push(todo);
+            }
             var listItem = document.createElement("li");
             listItem.textContent = todo.title;
             listItem.className = todo.complete ? "complete" : "incomplete";
 
-            var deleteButton = buttonFactory("delete", "delete-button", deleteEntry, todo);
+            var deleteButton = buttonFactory("delete", "button", deleteEntry, todo);
 
-            var completeButton = buttonFactory("complete", "complete-button", completeEntry, todo);
+            var completeButton = buttonFactory("complete", "button", completeEntry, todo);
 
             listItem.appendChild(deleteButton);
             listItem.appendChild(completeButton);
             todoList.appendChild(listItem);
         });
-        var classes = completeCounter.className.split(" ").filter(function(value) {
-            return value !== "hidden";
-        }).join(" ");
+        var classes = nameFilter(completeCounter);
         completeCounter.textContent = count;
         completeCounter.className = classes;
         if (count === 0) {
             completeCounter.className += " hidden";
         }
+        classes = nameFilter(deleteComplete);
+        deleteComplete.className = classes;
+        if (completeList.length === 0) {
+            deleteComplete.className += " hidden";
+        }
+        deleteComplete.onclick = function() {
+            deleteList(completeList);
+            return;
+        };
     });
 }
 
+function selector(value, mode) {
+    switch (mode){
+        case ALL_VALUES:
+            return true;
+        case ACTIVE_VALUES:
+            return !value.complete;
+        case COMPLETE_VALUES:
+            return value.complete;
+    }
+}
+
+function nameFilter(item) {
+    return item.className.split(" ").filter(function(value) {
+        return value !== "hidden";
+    }).join(" ");
+}
+
+function deleteList(todoList) {
+    for (var i = 0; i < todoList.length; i++) {
+        var createRequest = new XMLHttpRequest();
+        createRequest.open("DELETE", "/api/todo/" + todoList[i].id);
+        createRequest.setRequestHeader("Content-type", "application/json");
+        createRequest.onload = onLoadFactory(createRequest, "Failed to delete. Server returned ", 200,
+                               function() { return; });
+        createRequest.send();
+    }
+    reloadTodoList();
```

Comment:
Correct me if i'm wrong but doesn't this rely on luck to work? How do you know `reloadTodoList()` is running after all the deletes have been completed